### PR TITLE
add list and one meme view

### PR DESCRIPTION
**Pull request by kirillzykov**

### CODE REVIEW

File: client/package.json [hunk 1]
```diff
@@ -3,12 +3,14 @@
   "version": "0.1.0",
   "private": true,
   "dependencies": {
+    "@types/react-router-dom": "^5.1.6",
```
Comment (mikitachab) [inline client/package.json]: move to dev deps

File: client/package.json [hunk 2]
```diff
@@ -17,7 +19,7 @@
     "build": "react-scripts build",
     "test": "react-scripts test",
     "eject": "react-scripts eject",
-    "lint": "eslint . --ext .js,.jsx,.ts,.tsx --max-warnings=1"
+    "lint": "eslint . --ext .js,.jsx,.ts,.tsx --max-warnings=1 --fix"
```
Comment (mikitachab) [inline client/package.json]: rm --fix
create separate  script with --fix option

File: client/src/App.tsx [hunk 3]
```diff
@@ -1,19 +1,35 @@
 import React from 'react';
-import { Container, Navbar } from 'react-bootstrap';
+import { Container, Navbar, Nav } from 'react-bootstrap';
+import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
 import MemeForm from './components/MemeForm';
+import MemeList from './components/MemeList';
+import Meme from './components/Meme';
 import ServerHealthIndicator from './components/ServerHealthIndicator';
+import 'bootstrap/dist/css/bootstrap.min.css';
```
Comment (mikitachab) [inline client/src/App.tsx]: why we need this here?

File: client/src/components/Meme.tsx [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+import axios from 'axios';
+import { useParams } from 'react-router-dom';
+
+interface IMemeProp{
+  id: string,
+  name: string,
+  description: string,
+  tags: string[],
+  url: string,
+  //    date: Date,
```
Comment (mikitachab) [inline client/src/components/Meme.tsx]: why commented?

File: client/src/components/MemeLI.tsx [hunk 5]
```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+import Image from 'react-bootstrap/Image';
+//  import ListGroup from 'react-bootstrap/ListGroup';
+import { Col } from 'react-bootstrap';
+
+interface IMemeProp{
+  id: string,
+  name: string,
+  //    description: string,
```
Comment (mikitachab) [inline client/src/components/MemeLI.tsx]: delete or uncomment

File: client/src/components/MemeList.tsx [hunk 6]
```diff
@@ -0,0 +1,55 @@
+import React, { useEffect, useState } from 'react';
+import { Container, Row } from 'react-bootstrap';
+import axios from 'axios';
+import { useHistory } from 'react-router-dom';
+import MemeLI from './MemeLI';
+/* eslint no-underscore-dangle: 0 */
```
Comment (mikitachab) [inline client/src/components/MemeList.tsx]: ?

**Review by mikitachab — CHANGES_REQUESTED** [review]

File: client/src/components/MemeList.tsx [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+
+interface IMeme{
```
Comment (mikitachab) [inline client/src/components/MemeList.tsx]: there is too many similar "Meme" types

File: server/src/server.ts [hunk 8]
```diff
@@ -16,6 +18,8 @@ app.use(morgan('tiny')); // logging incoming request and response
 
 app.use('/api/v1/meme/uploads', uploadsRouter);
 app.use('/api/v1/meme/', memes);
+app.use('/api/v1/memes/', allMemesRouter);
```
Comment (mikitachab) [inline server/src/server.ts]: why we need separate router for every handler?

**Review by mikitachab — CHANGES_REQUESTED** [review]

File: client/src/components/MemeList.tsx [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+  const clickHandler = (id: string) => {
+    history.push(`/meme/${id}`);
+  };
+  return (
+    <Container>
+      <Row>
+        {
+          error ? 'response error'
+            : memes.map((meme) => (
+              <MemeLI
```
Comment (mikitachab) [inline client/src/components/MemeList.tsx:33]: this should be done in a way similar to:
```jsx
memes.map((meme, i) => (
<MemeLI
    meme={meme}
    onClick={clickHandler}
    key={i}
/>)
```

File: client/src/types/meme.interface.ts [hunk 10]
```diff
@@ -0,0 +1,8 @@
+export default interface IMeme{
```
Comment (mikitachab) [inline client/src/types/meme.interface.ts:1]: :+1:

File: server/src/server.ts [hunk 11]
```diff
@@ -15,7 +17,9 @@ app.use(cors());
 app.use(morgan('tiny')); // logging incoming request and response
 
 app.use('/api/v1/meme/uploads', uploadsRouter);
-app.use('/api/v1/meme/', memes);
+app.use('/api/v1/meme/', memes, allMemesRouter, oneMemeRouter);
+//  app.use('/api/v1/memes/', allMemesRouter);
```
Comment (mikitachab) [inline server/src/server.ts]: rm

**Review by mikitachab — CHANGES_REQUESTED** [review]

**Review by mikitachab — APPROVED** [review]

**Review by W2MANE — APPROVED** [review]

**Pull request merged**